vizro-ai/tests/unit/vizro-ai/dashboard/_response_models/test_components.py

```diff
@@ -0,0 +1,39 @@
+import vizro.models as vm
+from vizro_ai.dashboard._response_models.components import ComponentPlan
+
+
+class TestComponentCreate:
+    """Tests component creation."""
+
+    def test_component_plan_instantiation(self):
+        component = ComponentPlan(
+            component_id="card_1",
+            component_type="Card",
+            component_description="This is a card",
+            df_name="N/A",
+        )
+        assert component.component_id == "card_1"
+        assert component.component_type == "Card"
+        assert component.component_description == "This is a card"
+        assert component.df_name == "N/A"
+
+    def test_create_graph(self, mocker, component_plan_graph, mock_vizro_ai_return, df_metadata):
+        mock_vizro_ai_object = mocker.patch("vizro_ai.VizroAI.__init__")
+        mock_vizro_ai_object.return_value = None
+        mock_vizro_ai_call = mocker.patch("vizro_ai.VizroAI.plot")
+        mock_vizro_ai_call.return_value = mock_vizro_ai_return
+        result = component_plan_graph.create(
+            model=None,
+            all_df_metadata=df_metadata,
+        )
+        expected = vm.Graph(id="mock_id", figure=mock_vizro_ai_return)
+
+        assert result.dict(exclude={"id": True}) == expected.dict(exclude={"id": True})
+
```

Comment:
Shall we add one for AgGrid too? Happy to add it in the next PR though

Reply:
As agreed, I will add test for AgGrid in the next PR.